### PR TITLE
Added common utils for microsoft teams

### PR DESCRIPTION
**Pull request by dankyalo599**
### Description
Add common utils for microsoft teams
 
### Issues Resolved
Add class and unit tests for microsoft teams common utils
 
### Check List
- [X ] New functionality includes testing.
  - [X ] All tests pass
- [ X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by getsaurabh02** [comment]
Hi @dankyalo599 Are we still working on this PR? Looks like the workflow checks failed. Can we address this in the meantime while I get some attention on the review.

**Comment by lezzago** [comment]
There are ktlint errors that need to be fixed

**Comment by dankyalo599** [comment]
> Hi @dankyalo599 Are we still working on this PR? Looks like the workflow checks failed. Can we address this in the meantime while I get some attention on the review.

Yes we are still working on this PR

File: src/main/kotlin/org/opensearch/commons/notifications/model/MicrosoftTeams.kt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+data class MicrosoftTeams(
+        val url: String
+=======
+/**
+ * Data class representing Microsoft Teams channel.
+ */
+data class MicrosoftTeams(
+    val url: String
+>>>>>>> b13bf3d337f15337a3c6999c5285482a79fd5a5d
```
Comment (dblock) [inline src/main/kotlin/org/opensearch/commons/notifications/model/MicrosoftTeams.kt:33]: Looks like you have code merge conflicts here.

**Comment by zhichao-aws** [comment]
Hi, I create a new [PR](https://github.com/opensearch-project/common-utils/pull/428) to fix the build and add more test case.  PR in notifications: https://github.com/opensearch-project/notifications/pull/676

**Comment by engechas** [comment]
Looks like this was covered by #428. Closing, please reopen if I misunderstood

**Pull request closed without merging**